ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -288,15 +308,19 @@ localRoots actions@PeerSelectionActions{ readLocalRootPeers
                                        }
            policy
            st@PeerSelectionState{
-             bigLedgerPeers,
              localRootPeers,
              publicRootPeers,
              knownPeers,
              establishedPeers,
              activePeers,
              inProgressDemoteHot,
-             targets = PeerSelectionTargets{targetNumberOfKnownPeers}
-           } =
+             targets = PeerSelectionTargets{targetNumberOfKnownPeers},
+             currentLedgerStateJudgement
+           }
+  -- If the current ledger state is TooOld we can only trust our local
+  -- root peers, this means that if we remove any local root peer we
+  -- no longer
+  | currentLedgerStateJudgement /= TooOld =
```

Comment:
I think the SPO should just restart the node at this point and probably reset its chain cache, no? Also if IIUC it could be that we go to a `YoungEnough` state during syncing, we should document that `TooOld` state is a sensitive and vulnerable state and so making dynamic changes to topology file is disallowed and if one thinks their topology is misconfigured should take the right precautions and restart the node

Reply:
It is quite surprising that local roots will not change in bootstrap phase, so we'd need a strong argument why this is disallowed - and I don't see why we need this in the first place.

Reply:
In bootstrap phase we should _only_ be connected to trusted peers. When the node detects a `TooOld` state change, it will discard all connections and start fresh from only trusted peers.

Local roots might have non trusted peers. If we allow local roots reconfiguration whilst in `TooOld` state this can lead to invariant breaking and we'd have to purge all the connections again and make sure we only use trusted peers. This could be done, but I don't see a compelling argument as to why we should embrace providing this feature for such a sensitive state. It is easier to tell users that if the node is in `TooOld` state it is considered a sensitive state so the node will only be connected to trusted peers - if they realized local trusted peers are no longer to be trusted than they started from false premises and should restart everything, including their chain cache